nilearn/image/image.py

```diff
@@ -205,18 +208,25 @@ def _smooth_array(arr, affine, fwhm=None, ensure_finite=True, copy=True):
         # SPM tends to put NaNs in the data outside the brain
         arr[np.logical_not(np.isfinite(arr))] = 0
 
-    if fwhm == 'fast':
+    if isinstance(fwhm, str) and (fwhm == 'fast'):
         arr = _fast_smooth_array(arr)
     elif fwhm is not None:
+        # Assume fwhm is either a nonzero number or a tuple, or a numpy array.
+        # Whatever it is at this point, transform it into an np.array
+        fwhm = np.asarray(fwhm)
+        fwhm = np.where(fwhm == None, 0.0, fwhm)
```

Comment:
```python
fwhm = np.asarray(fwhm, dtype=float)
fwhm = np.nan_to_num(fwhm)
```

this will avoid `fwhm` 's dtype to be "object" if it contained None, (and avoids flake8 complaining about the comparison to None with ==)

Reply:
What Jerome said.

For general reference, equivalence with `None` should always be tested with `is`, not `==`

Reply:
here fwhm is a numpy array so `==` has a different meaning and performs element-wise comparison. `fwhm is None` would not give what we need. still I would prefer `dtype=float` and `nan_to_num` to get an array of the right dtype

Reply:
> so == has a different meaning and performs element-wise comparison. fwhm is None would not give what we need.

Did not know that that. Thanks @pausz & @jeromedockes